specifyweb/frontend/js_src/lib/components/TreeView/index.tsx

```diff
@@ -118,50 +98,63 @@ function TreeView<SCHEMA extends AnyTree>({
     [baseUrl, sortField]
   );
 
-  const statsThreshold = getPref(
-    `TreeEditor.Rank.Threshold.${tableName as 'Geography'}`
-  );
-  const getStats = React.useCallback(
-    async (nodeId: number | 'null', rankId: number): Promise<Stats> =>
-      rankId >= statsThreshold
-        ? fetchStats(`${baseUrl}/${nodeId}/stats/`)
-        : Promise.resolve({}),
-    [baseUrl, statsThreshold]
-  );
-
   const [rows, setRows] = useAsyncState<RA<Row>>(
     React.useCallback(async () => getRows('null'), [getRows]),
     true
   );
-  const id = useId('tree-view');
 
   // FEATURE: synchronize focus path with the URL
-  const [focusPath = [], setFocusPath] = useCachedState(
+  const [_focusedPath = [], setFocusPath] = useCachedState(
     'tree',
     `focusPath${tableName}`
   );
-  const [focusedRow, setFocusedRow] = React.useState<Row | undefined>(
-    undefined
-  );
+  const [focusPathTop = [], setFocusPathTop] = React.useState<RA<number>>();
+  const [focusPathBottom = [], setFocusPathBottom] =
```

Comment:
focus path top and bottom should probably be moved into useTreeStates too

Reply:
the hook's return type could look something like this:

```ts
RR<
  'top' | 'bottom',
  {
    readonly focusedRow: GetOrSet<Row | undefined>;
    readonly focusPath: GetOrSet<RA<number>>;
  }
>
```

and then later in the file where you have:

```ts
          isTop
            ? [focusPathTop, setFocusPathTop]
            : [focusPathBottom, setFocusPathBottom]
```

replace by:
```ts
   treeStates[type].focusPath
```

Reply:
also, whenever setFocusPathTop is called, setFocusPath should be called too (so that local storage is updated with new path


<details><summary>Hint</summary>
<p>

 to accommodate that, move the useCachedState hook inside the useTreeStates

</p>
</details> 

<details><summary>Solution</summary>
<p>

 to accommodate that, move the useCachedState hook inside the useTreeStates, then that hook instead of returning setFocusPathTop, will return a React.useCallback, which in turn will call both setFocusPathTop and setFocusPath (note variable names would be different in that case - I am using the variable names you have here for now - i.e "Top" won't make sense inside the useTreeStates)

</p>
</details>